src/broadcast_stage.rs

```diff
@@ -180,11 +182,20 @@ impl BroadcastStage {
         entry_height: u64,
         recycler: BlobRecycler,
         receiver: BlobReceiver,
+        voting_nodes: Arc<RwLock<VotingNodes>>,
     ) -> Self {
         let thread_hdl = Builder::new()
             .name("solana-broadcaster".to_string())
             .spawn(move || {
-                Self::run(&sock, &crdt, &window, entry_height, &recycler, &receiver);
+                Self::run(
```

Comment:
this pattern of more parameters whenever we refactor worries me a little, I don't know how to combat it right now, or whether we should, but it itches at me. a refactor doesn't make things like this move in the opposite direction?